### PR TITLE
Hyacinth 492 filter type

### PR DESCRIPTION
**Pull request by barmintor**

### CODE REVIEW

File: app/graphql/types/extensions/map_to_digital_objects.rb [hunk 1]
```diff
@@ -8,8 +8,8 @@ def resolve(object:, arguments:, context:)
       end
 
       def after_resolve(object:, value:, arguments:, context:, memo:)
-        raise GraphQL::ExecutionError, 'ToDigitalObjects can only be downstream of Paginate' unless value.is_a?(OpenStruct) && value[:page_info].is_a?(OpenStruct)
-        value[:nodes] = value[:nodes].map { |digital_object_record| ::DigitalObject::Base.find(digital_object_record.uid) }
+        raise GraphQL::ExecutionError, 'ToDigitalObjects can only be downstream of SolrSearch' unless value.is_a?(OpenStruct) && value[:page_info].is_a?(OpenStruct)
```
Comment (elohanlon) [inline app/graphql/types/extensions/map_to_digital_objects.rb]: @barmintor did you mean "MapToDigitalObjects" here? Or is the "Map" part not made visible to the client in the way that it's used?

Comment (barmintor) [reply app/graphql/types/extensions/map_to_digital_objects.rb]: really just an oversight

File: lib/solr/params.rb [hunk 2]
```diff
@@ -9,6 +9,7 @@ def initialize
         q: nil,
         qt: 'search',
         fq: [],
+        'facet.field'.to_sym => [],
         # rows: URIService::DEFAULT_LIMIT,
```
Comment (elohanlon) [inline lib/solr/params.rb]: Okay to delete this comment?

File: lib/tasks/hyacinth.rake [hunk 3]
```diff
@@ -1,4 +1,10 @@
 # frozen_string_literal: true
 
 namespace :hyacinth do
+  task reindex: :environment do
+    DigitalObjectRecord.find_in_batches(batch_size: 100) do |records|
```
Comment (elohanlon) [inline lib/tasks/hyacinth.rake]: @barmintor We could probably bump this up to `batch_size: 1000` if you and @cgalarza don't mind.

File: spec/fixtures/solr_cores/hyacinth-solr-6-3/conf/solrconfig.xml [hunk 4]
```diff
@@ -748,13 +748,33 @@
...
     -->
+  <!-- request handler to return searches -->
+  <requestHandler name="/search" class="solr.SearchHandler">
+    <lst name="defaults">
+      <str name="echoParams">explicit</str>
+      <str name="defType">edismax</str>
+      <str name="q">*:*</str>
+      <str name="q.alt">*:*</str>
+      <str name="rows">10</str>
+      <str name="qf">id^500, digital_object_type_ssi^40 title_ssi^30</str>
```
Comment (elohanlon) [inline spec/fixtures/solr_cores/hyacinth-solr-6-3/conf/solrconfig.xml]: Why is digital_object_type_ssi in here?  I don't remember if this is a thing carried over from Hyacinth 2, and if so, why we're including it there.

Comment (barmintor) [reply spec/fixtures/solr_cores/hyacinth-solr-6-3/conf/solrconfig.xml]: those are just the existing fields in a proposed score

Comment (barmintor) [reply spec/fixtures/solr_cores/hyacinth-solr-6-3/conf/solrconfig.xml]:
```suggestion
      <str name="qf">id^500, title_ssi^30</str>
```

File: lib/tasks/hyacinth.rake (hunk 3 above)
Comment (barmintor) [reply lib/tasks/hyacinth.rake]:
```suggestion
    DigitalObjectRecord.find_in_batches(batch_size: (ENV['BATCH_SIZE'] || 1000).to_i) do |records|
```

File: lib/solr/params.rb (hunk 2 above)
Comment (barmintor) [reply lib/solr/params.rb]:
```suggestion
```

File: app/graphql/types/extensions/map_to_digital_objects.rb (hunk 1 above)
Comment (barmintor) [reply app/graphql/types/extensions/map_to_digital_objects.rb]:
```suggestion
        raise GraphQL::ExecutionError, 'MapToDigitalObjects can only be downstream of SolrSearch' unless value.is_a?(OpenStruct) && value[:page_info].is_a?(OpenStruct)
```

File: spec/hyacinth/adapters/digital_object_search_adapter/solr_spec.rb [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+      expect(adapter.solr_document_for(authorized_object)['id']).to eql(authorized_object.uid)
+    end
+
+    it "indexes the digital object type" do
+      expect(adapter.solr_document_for(authorized_object)['digital_object_type_ssi']).to eql(authorized_object.digital_object_type)
+    end
+
+    it "indexes the title into title_ssi" do
+      pending("object factory field name corrections to align with title practice")
+      expect(adapter.solr_document_for(authorized_object)['title_ssi']).to eql('Tall Man and His Hat')
```
Comment (elohanlon) [inline spec/hyacinth/adapters/digital_object_search_adapter/solr_spec.rb]: Should this be 'The Tall Man and His Hat' (including non-sort portion)?  Or is title_ssi meant to be the sort-only portion of the title?

Comment (elohanlon) [reply spec/hyacinth/adapters/digital_object_search_adapter/solr_spec.rb]: Never mind.  Missed that this is a pending test.

File: spec/models/concerns/digital_object_concerns/index_behavior_spec.rb [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  context "#save" do
+    let!(:prior) do
+      results = Hyacinth::Config.digital_object_search_adapter.search(search_params)
+      results['response']['docs'].first
+    end
+    it "runs as expected, updating the saved object in the index" do
+      digital_object_with_sample_data.save
+      results = Hyacinth::Config.digital_object_search_adapter.search(search_params)
+      curr = results['response']['docs'].first
+      puts(results.inspect)
```
Comment (elohanlon) [inline spec/models/concerns/digital_object_concerns/index_behavior_spec.rb]: Leftover `puts` here.

Comment (barmintor) [reply spec/models/concerns/digital_object_concerns/index_behavior_spec.rb]:
```suggestion
```

File: lib/tasks/hyacinth.rake (hunk 3 above)
Comment (barmintor) [reply lib/tasks/hyacinth.rake]: How do you feel about the suggested change?

Comment (elohanlon) [reply lib/tasks/hyacinth.rake]: Perfect! 👍

File: app/graphql/types/pageable.rb [hunk 7]
```diff
@@ -16,6 +16,7 @@ def results_type(&block)
 
           field :totalCount, !types.Int, "Count of total results", property: :total_count
           field :nodes, types[wrapped_type], "A list of nodes."
+          field :facets, types[Types::Facets::FieldType], "A list of facets.", property: :facets
```
Comment (cgalarza) [inline app/graphql/types/pageable.rb]: Are we okay with saying all paginatable responses will have a `facets` field knowing that objects that are backed by the db won't have this ability?

File: lib/solr/params.rb [hunk 8]
```diff
@@ -19,6 +19,11 @@ def fq(field, value)
       self
     end
 
+    def facet_on(field)
+      @parameters[:"facet.field"] << field unless field.blank?
```
Comment (cgalarza) [inline lib/solr/params.rb]:
```suggestion
      @parameters["facet"] = "on"
      @parameters[:"facet.field"] << field unless field.blank?
```
Does this simplify the `to_h` method? If not feel free to ignore.

File: app/graphql/types/pageable.rb (hunk 7 above)
Comment (barmintor) [reply app/graphql/types/pageable.rb]: hmm I guess I could move that field into the block param, or make it conditional based on a param. Do you have a preference?

Comment (cgalarza) [reply app/graphql/types/pageable.rb]: I guess i was thinking making it conditional based on a param or defining two types `results_type` and `results_with_facet_type`. 

I do think this is work that can be ticketed for later tho if you want. I am going to approve the PR.

Comment (barmintor) [reply app/graphql/types/pageable.rb]: ok, sounds good

**Pull request merged**